docs/fundamentals/networking/sockets/tcp-classes.md

```diff
@@ -78,10 +78,191 @@ The preceding C# code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+```
+
+### Start listening on the server
+
+The <xref:System.Net.Sockets.TcpListener.Start> method is a wrapper combining Socket's <xref:System.Net.Sockets.Socket.Bind%2A> and <xref:System.Net.Sockets.Socket.Listen>.
+
+Consider the following TCP listener code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+listener.Start(10);
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+socket.Bind(ep);
+try
+{
+    socket.Listen(10);
+}
+catch (SocketException)
+{
+    socket.Dispose();
+}
+```
+
+### Accepting a connection on the server
+
+`TcpListener` can accept both <xref:System.Net.Sockets.Socket> (via <xref:System.Net.Sockets.TcpListener.AcceptSocket> or <xref:System.Net.Sockets.TcpListener.AcceptSocketAsync>) and <xref:System.Net.Sockets.TcpClient> (via <xref:System.Net.Sockets.TcpListener.AcceptTcpClient> and <xref:System.Net.Sockets.TcpListener.AcceptTcpClientAsync>).
+
+Consider the following TCP client code:
+
+```csharp
+var listener = new TcpListener(IPAddress.Loopback, 5000);
+var acceptSocket = listener.AcceptSocket(); // var acceptSocket = listener.AcceptSocketAsync();
+```
+
+The preceding TCP listener code is functionally equivalent to the following socket code:
+
+```csharp
+var ep = new IPEndPoint(IPAddress.Loopback, 5000);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+var acceptSocket = socket.Accept(); // var acceptSocket = await socket.AcceptAsync();
+```
+
+### Create a `NetworkStream` to send and receive data
+
+`TcpClient` class has a <xref:System.Net.Sockets.TcpClient.GetStream> helper method to instantiate a <xref:System.Net.Sockets.NetworkStream>. With `Socket`, you have to do it manually.
+
+Consider the following `TcpClient` code:
+
+```csharp
+var client = new TcpClient();
+NetworkStream stream = client.GetStream();
+```
+
+Which is equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+NetworkStream stream = new(socket, ownsSocket: true); // Please be aware, we're passing true as `ownsSocket` parameter to let stream object to own the socket. 
+```
+
+> [!TIP]
+> <xref:System.Net.Sockets.Socket> class has advantage if you don't want to use `Stream`. There are <xref:System.Net.Sockets.Socket.Send%2A>, <xref:System.Net.Sockets.Socket.SendAsync%2A>, <xref:System.Net.Sockets.Socket.Receive%2A> and <xref:System.Net.Sockets.Socket.ReceiveAsync%2A> overloads in `Socket` class.
```

Comment:
```suggestion
> If your code doesn't need to work with a <xref:System.IO.Stream> instance, you can rely on `Socket`'s Send/Receive methods (<xref:System.Net.Sockets.Socket.Send%2A>, <xref:System.Net.Sockets.Socket.SendAsync%2A>, <xref:System.Net.Sockets.Socket.Receive%2A> and <xref:System.Net.Sockets.Socket.ReceiveAsync%2A>) directly instead of creating a <xref:System.Net.Sockets.NetworkStream>.
```